xdsl/dialects/scf.py

```diff
@@ -69,6 +69,11 @@ def verify_(self):
                 f"{len(self.body.block.args)}. The body must have the induction "
                 f"variable and loop-carried variables as arguments."
             )
+        if self.body.block.args and (iter_var := self.body.block.args[0]):
+            if not isinstance(iter_var.typ, IndexType):
+                raise VerifyException(
+                    f"Type {iter_var.typ} of body first argument is not an index for the induction variable"
```

Comment:
```suggestion
                    f"The first block argument of the body is of type {iter_var.typ} instead of index"
```

Maybe this is a bit clearer?